app/features/settings/settings-sidebar-provider.tsx

```diff
@@ -0,0 +1,64 @@
+import { type ReactNode, createContext, useContext, useState } from 'react';
+import type {
+  NavigateToView,
+  SettingsSidebarState,
+  SettingsView,
+} from './types';
+
+interface SettingsSidebarContextType {
+  state: SettingsSidebarState;
+  navigateToView: NavigateToView;
+}
+
+interface SettingsSidebarProviderProps {
+  children: ReactNode;
+}
+
+const SettingsSidebarContext = createContext<
+  SettingsSidebarContextType | undefined
+>(undefined);
+
+const defaultState: SettingsSidebarState = {
+  view: 'main',
+};
+
+export function SettingsSidebarProvider({
```

Comment:
should we just add zustand and use that to manage state instead of having to define our own context? it's less boilerplaty and will probably be useful for other stuff as well

Reply:
yea zustand looks cool, I added it to the animated-navigation in #216